geode-core/src/main/java/org/apache/geode/management/internal/FederatingManager.java

```diff
@@ -100,6 +100,10 @@ public FederatingManager(MBeanJMXAdapter jmxAdapter, ManagementResourceRepo repo
     this.messenger = new MemberMessenger(jmxAdapter, repo, system);
   }
 
+  void setProxyFactory(MBeanProxyFactory newProxyFactory) {
```

Comment:
I'm still not real happy that we had to add this method to product code be able to test to allow mocking/spying MBeanProxyFactory for unit testing. I think that we should add the `@TestingOnly` annotation to the method to highlight that it is only intended to be used in tests.

Reply:
Good idea. I'll add that now. I don't think there is a way around adding this method for the test though.

Reply:
I agree that it's needed for exposing the original problem in a deterministic test. At least adding the annotation won't affect the test runs.